Automatically modify read or base counts according shared_key set

vladsaveliev opened the pull request:
Will make the read and base counts in custom content affected by modifiers. 

The problem is that we can't assign lambdas in YAML, thus can't set proper `modify` value for custom content. Making use of `shared_key` to automatically set the modification would be nice, though the downside is that it makes it implicit and also breaks compatibility, which is probably not desired. So rather opening to start discussion.

vladsaveliev commented:
Just remembered that I already created a related issue https://github.com/ewels/MultiQC/issues/470! So can continue moving from here.

ewels commented:
I think this looks good and I'm happy to merge as a first step if you don't intend to add anything else imminently.

What do you think about setting `min` and `format` automatically in the same blocks if they're not already set?

vladsaveliev commented:
Absolutely. Happy to add anything you think is reasonable. I guess everything from your list?
* `title`: prepend `config.read_count_prefix`
* `description`: add `({})` suffix
* `min`: set to `0`
* `modify`: set to `lambda x: x * config.read_count_multiplier`
* `format`: `'{:,.2f} ' + config.read_count_prefix if config.read_count_multiplier != 1 else read_format = '{:,.0f}'`

ewels commented:
Yeah - though `title` `description` are a little more controversial as we're adding to something that's already set. I was thinking that for the others we'd only set if not already set (so easy to override). But as long as we go through all of the core modules updating them so that the new code works it should be fine (a bit of clever regex copy and paste perhaps?).

Phil

ewels commented:
ps. I reran the failed travis job and it worked this time 👍

vladsaveliev commented:
Actually cannot agree more with the concerns on `title`, since in my reports I add `M` as a `suffix` rather than to the header. Often it appears to be more clear, but primarily it allows to save horizontal space since the column header is usually wider than the values.

vladsaveliev commented:
Added `min` and `format`, moved that code to a more proper location (before the header key defaults are set), and modified modules `qualimap` and `bcl2fastq` accordingly for testing and to start cleaning up modules. Can be merged as a first step.

ewels commented on multiqc/modules/qualimap/QM_BamQC.py:
```diff
@@ -505,7 +505,6 @@ def general_stats_headers (self):
         'title': 'Ins. size',
         'description': 'Median insert size',
         'min': 0,
-        'suffix': 'bp',
```
Sorry - tiny thing: removing this is fine, but please add back to the description in that case so that the info is still there

ewels commented:
Brilliant stuff, thanks!
